src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSourceCredential.cs

```diff
@@ -185,5 +186,46 @@ private static IEnumerable<string> ParseAuthTypeFilterString(string str)
                 ? Enumerable.Empty<string>()
                 : str.Split(',').Select(x => x.Trim()).Where(x => x != string.Empty);
         }
+
+
+        public CredentialsItem AsCredentialsItem()
+        {
+            return new CredentialsItem(Source, Username, PasswordText, IsPasswordClearText, ValidAuthenticationTypesText);
+        }
+
+        public bool Equals(PackageSourceCredential other)
+        {
+            if (other == null)
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            return string.Equals(Source, other.Source, StringComparison.Ordinal) &&
+                string.Equals(Username, other.Username, StringComparison.Ordinal) &&
+                string.Equals(PasswordText, other.PasswordText, StringComparison.Ordinal) &&
+                IsPasswordClearText == other.IsPasswordClearText;
+        }
+
+        public override bool Equals(object other)
+        {
+            return Equals(other as PackageSourceCredential);
+        }
+
+        public override int GetHashCode()
+        {
+            var combiner = new HashCodeCombiner();
+
+            combiner.AddObject(Source);
+            combiner.AddObject(Username);
+            combiner.AddObject(PasswordText);
+            combiner.AddObject(IsPasswordClearText);
+
+            return combiner.GetHashCode();
```

Comment:
I was following the same behavior we have throughout the code where we don't memoize it, but I think it might be a good idea to memoize this since the type is immutable